### PR TITLE
Polyhedron Demo: .mesh reading enhancement

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
When reading a .mesh in the demo, if a subdomain is 0, make it 1.

### CODE REVIEW

File: Mesh_3/include/CGAL/Mesh_3/tet_soup_to_c3t3.h [hunk 1]
```diff
@@ -141,7 +151,15 @@ bool build_finite_cells(Tr& tr,
 
     Cell_handle c = tr.tds().create_cell(vs[0], vs[1], vs[2], vs[3]);
     c->set_subdomain_index(tet[4]); // the cell's info keeps the reference of the tetrahedron
-
+    c->info() = tet[4]; // the cell's info keeps the reference of the tetrahedron
```
Comment (janetournois) [inline Mesh_3/include/CGAL/Mesh_3/tet_soup_to_c3t3.h:154]: tetrahedral remeshing uses this function, and we do not have `info()` in cells.
Why isn't `subdomain_index()` enough?

**Comment by janetournois** [comment]
Can you please explain why you make a 0 subdomain_index 1?
0 is for "outside"... it status should not be changed, right?

**Comment by sloriot** [comment]
If you read random medit file 0 is not a special value

**Comment by janetournois** [comment]
However it is special or not, why do we change it?
It amounts to merging 2 subdomains... (number 0 and number 1)

**Comment by sloriot** [comment]
It is done only if 1 is not present IIRC

File: Mesh_3/include/CGAL/Mesh_3/tet_soup_to_c3t3.h (hunk 1 above)
Comment (maxGimeno) [reply Mesh_3/include/CGAL/Mesh_3/tet_soup_to_c3t3.h]: It seems it is after all, must have been a reminiscence from an older try

**Comment by maxGimeno** [comment]
> However it is special or not, why do we change it?
> It amounts to merging 2 subdomains... (number 0 and number 1)

It doesn't. It takes the first number that is not used by subdomains

**Comment by janetournois** [comment]
Ok.
Two more questions : 
- what if you have negative subdomain indices?
- `prevent_domain_0 = true` sounds like the subdomain 0 is avoided by default. From what I understand, it is used the over way round. Am I wrong?

**Comment by maxGimeno** [comment]
I did not change anything about the negative subdomain indices.

In the case where prevent_domain_0 is true,  there is a CGAL_precondition(tet[4] > 0)), so the domain 0 is forbidden in the .mesh file. That is why I call it `prevent`.

**Comment by maxGimeno** [comment]
errors https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.2-Ic-2/Tetrahedral_remeshing_Examples/TestReport_lrineau_ArchLinux-clang.gz

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.2-Ic-3.shtml

**Pull request merged**